pkg/image/controller/imagestream_controller.go

```diff
@@ -116,14 +116,21 @@ func (c *ImageStreamController) processNextWorkItem() bool {
 	}
 	defer c.queue.Done(key)
 
-	stream, err := c.getByKey(key.(string))
+	namespace, name, err := cache.SplitMetaNamespaceKey(key.(string))
 	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("Error syncing image stream: %v", err))
+		glog.V(2).Infof("Unable to split namespace key for key %q: %v", key, err)
 		c.queue.AddRateLimited(key)
 		return true
 	}
-	if stream == nil {
-		c.queue.Forget(key)
+	stream, err := c.lister.ImageStreams(namespace).Get(name)
```

Comment:
Can we reshape this like upstream, where the sync function takes the key and does the work there?  it keeps the error handling code a little cleaner and more standard.